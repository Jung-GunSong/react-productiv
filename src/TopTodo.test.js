import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";

const testTodo = [{
  id: 1,
  title: "test1",
  description: "testing1",
  priority: 2,
},
{
  id: 2,
  title: "test2",
  description: "testing testing",
  priority: 1,
},
{
  id: 3,
  title: "test3",
  description: "still testing....",
  priority: 2,
}];

describe("TopTodo component", function () {
  it("renders without crashing", function () {
    render(<TopTodo todos={testTodo}/>);
  });

  it("matches snapshot", function () {
    const {container} = render(<TopTodo todos={testTodo}/>);
    expect(container).toMatchSnapshot();
  });

  it("renders top todo if todos present", function () {
    const {getByText} = render(<TopTodo todos={testTodo}/>);
    expect(getByText("test2")).toBeInTheDocument();
    expect(getByText("testing testing")).toBeInTheDocument();
    expect(getByText("priority: 1")).toBeInTheDocument();
  });


});
