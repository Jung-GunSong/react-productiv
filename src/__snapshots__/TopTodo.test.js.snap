// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTodo component matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        test2
      </b>
       
      <small>
        priority: 
        1
      </small>
    </div>
    <div>
      <small>
        testing testing
      </small>
    </div>
  </div>
</div>
`;
